x/update_validators/init.go

```diff
@@ -0,0 +1,36 @@
+package update_validators
+
+import (
+	"github.com/confio/weave"
+)
+
+const optKey = "update_validators"
+
+// Initializer fulfils the InitStater interface to load data from
+// the genesis file
+type Initializer struct{}
+
+var _ weave.Initializer = Initializer{}
+
+// FromGenesis will parse initial account info from genesis
+// and save it to the database
+func (Initializer) FromGenesis(opts weave.Options, kv weave.KVStore) error {
+	accounts := WeaveAccounts{}
+	err := opts.ReadOptions(optKey, &accounts)
+	if err != nil {
+		return err
+	}
+	bucket := NewBucket()
+	for _, addr := range accounts.Addresses {
+		if err := addr.Validate(); err != nil {
+			return err
+		}
+		wallet := AccountsWith(accounts)
```

Comment:
But here, you create a wallet with *all account* not just the one you checked... 

This only makes sense if you end the for loop on the line above.

Reply:
Woops, for loop is not needed here at all.